specification/search/data-plane/Azure.Search/preview/2025-05-01-preview/searchservice.json

```diff
@@ -0,0 +1,14715 @@
+              "multimodalContent": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "type": {
+                      "type": "string",
+                      "description": "The type of the message content."
+                    },
+                    "text": {
+                      "type": "string",
+                      "description": "The text of a string message content."
+                    },
+                    "image": {
+                      "type": "object",
+                      "properties": {
+                        "url": {
+                          "type": "string",
+                          "description": "The url of the image"
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            "required": [ "role" ]
+          }
+        },
+        "includeReferenceSourceData": {
+          "type": "boolean",
+          "description": "Whether to include the source data in the response references."
+        }
+      },
+      "description": "The input contract for the retrieval request."
+    },
+    "KnowledgeAgentRetrievalResponse": {
+      "type": "object",
+      "properties": {
+        "response": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "role": {
+                "type": "string",
+                "description": "The role of the tool response."
+              },
+              "content": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "type": {
+                      "type": "string",
+                      "description": "The type of the message content."
+                    },
+                    "text": {
+                      "type": "string",
+                      "description": "The text of a string message content."
+                    },
+                    "image": {
+                      "type": "object",
+                      "properties": {
+                        "url": {
+                          "type": "string",
+                          "description": "The url of the image"
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            "required": [ "role", "content" ]
+          }
+        },
+        "activity": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentActivityRecord"
+          },
+          "description": "The activity records for tracking progress and billing implications."
+        },
+        "references": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentReference"
```

Comment:
Similar question here as activity record as there are multiple types for reference but there isn't a way to discriminator what type they are.